src/snowflake/snowpark/session.py

```diff
@@ -869,81 +904,69 @@ def _resolve_packages(
+            SNOWPARK_PACKAGE_NAME
+        ]
         valid_packages = (
             {
                 p[0]: json.loads(p[1])
                 for p in self.table(package_table)
                 .filter(
                     (col("language") == "python")
-                    & (col("package_name").in_([v[0] for v in package_dict.values()]))
+                    & (col("package_name").in_(package_search_list))
                 )
                 .group_by("package_name")
                 .agg(array_agg("version"))
                 ._internal_collect_with_tag()
             }
-            if validate_package and package_dict
+            if len(package_search_list) > 0
             else None
         )
 
         result_dict = (
             existing_packages_dict if existing_packages_dict is not None else {}
         )
+        unsupported_packages = []
         for package, package_info in package_dict.items():
             package_name, use_local_version, package_req = package_info
             package_version_req = package_req.specs[0][1] if package_req.specs else None
 
             if validate_package:
-                unavailable_pkg_err_msg = (
-                    "it is not available in Snowflake. Check information_schema.packages "
-                    "to see available packages for UDFs. If this package is a "
-                    '"pure-Python" package, you can find the directory of this package '
-                    "and add it via session.add_import(). See details at "
-                    "https://docs.snowflake.com/en/developer-guide/snowpark/python/creating-udfs.html#using-third-party-packages-from-anaconda-in-a-udf."
-                )
-
-                if package_name not in valid_packages:
-                    is_anaconda_terms_acknowledged = self._run_query(
-                        "select system$are_anaconda_terms_acknowledged()"
-                    )[0][0]
-                    if is_anaconda_terms_acknowledged:
-                        detailed_err_msg = unavailable_pkg_err_msg
+                if package_name not in valid_packages or (
+                    package_version_req
+                    and not any(v in package_req for v in valid_packages[package_name])
+                ):
+                    if self._is_anaconda_terms_acknowledged():
+                        unsupported_packages.append(package)
+                        continue
                     else:
-                        detailed_err_msg = (
-                            "Anaconda terms must be accepted by ORGADMIN to use "
-                            "Anaconda 3rd party packages. Please follow the instructions at "
+                        version_text = (
+                            f"(version {package_version_req})"
+                            if package_version_req is not None
+                            else ""
+                        )
+                        raise ValueError(
```

Comment:
Maybe `RuntimeError`? I'm not sure if `ValueError` is the most appropriate here.